### PR TITLE
#Refactor code

### PR DESCRIPTION
**Pull request by Nejjari001**
- Ensured YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Followed SOLID principles.
- Ensured that the role of main.rb is only to start the code and ask the user for the initial choice of options (or to quit).
- Extracted some pieces of logic from main.rb to menu.rb file

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -1,130 +1,145 @@
...
+
+      @rentals.each do |rental|
+        if rental.person.id == id
+          puts "Book Name: #{rental.book.title} Book Author: #{rental.book.author} Date: #{rental.date}"
+        else
+          puts "No book found by person id #{id}"
+        end
+      end
+    else
+      puts 'There are no rentals'
```
Comment (danmainah) [inline app.rb]: - [ ] @Nejjari001 and @Nowembabazi great work on refactoring however its best practice to ensure all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input.

Comment (danmainah) [reply app.rb]: - [ ] Also check on your rental display method got the response below when running the app, showing I have no rentals but still displaying the rentals 
image
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Nejjari001 and @Nowembabazi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by roshan-bajgain — APPROVED** [review]
Hi @Nejjari001, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️

## Optional suggestions
- [ ] Everything is good but here is a small issue please fix it. while creating a student and selecting parent permission I got an error. Please see below.
- [ ] You can fix this issue by removing the true from `parent_permission: true` in person.rb.
### Error
error

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**